x-pack/platform/plugins/shared/fleet/server/routes/data_streams/handlers.ts

```diff
@@ -257,3 +257,107 @@ export const getListHandler: RequestHandler = async (context, request, response)
     throw err;
   }
 };
+
+export const getDeprecatedILMCheckHandler: RequestHandler = async (context, request, response) => {
```

Comment:
we should do an early return if [disableILMPolicies](https://github.com/elastic/kibana/blob/5456472f823055cd18617c59bb3fe5497e593c6b/x-pack/platform/plugins/shared/fleet/server/services/epm/elasticsearch/template/default_settings.ts#L21) config is false, it is set in serverless where the ILM policies are not available